app/ide-desktop/lib/icons/src/index.js

```
@@ -178,9 +179,11 @@ async function genIcons() {
             .toFile(`${distPath}/png/${outName}`)
     }
 
-    console.log('Generating ICNS.')
-    exec(`cp -R ${distPath}/png ${distPath}/png.iconset`)
-    exec(`iconutil --convert icns --output ${distPath}/icon.icns ${distPath}/png.iconset`)
+    if (platform() === 'darwin') {
```

Does this need to be gated for MacOS? Why did it not need to be gated before?

---

Good catch! I believe it does not have to be gated and thus it allows generating the package for macOS on non-macros platforms with icons. No one is using this functionality now, but I would leave it non-gated if its possible.

---

@MichaelMauderer @wdanilo No, it works only on macOS. Linux does not provide `iconutil`, not to mention of `cp -R` on Windows.

>  Why did it not need to be gated before?

Previously the command was just trying to asynchronously spawn child process but was silently ignoring any errors. Now it is checked.